### PR TITLE
executor: Parallelly run table lookup tasks

### PR DESCRIPTION
**Pull request by shenli**
@coocood

### CODE REVIEW

File: executor/executor_xapi.go [hunk 1]
```diff
@@ -210,23 +221,12 @@ func (e *XSelectIndexExec) Next() (*Row, error) {
 		task := e.tasks[e.taskCursor]
 		if !task.done {
```
Comment (coocood) [inline executor/executor_xapi.go]: need lock

File: executor/executor_xapi.go [hunk 2]
```diff
@@ -240,14 +240,33 @@ func (e *XSelectIndexExec) Next() (*Row, error) {
...
+			}
+		}
+		e.mu.Unlock()
+		if task == nil {
+			// No more task to run.
+			break
+		}
+		// Execute the picked task.
+		err := e.executeTask(task)
+		task.status = taskDone
```
Comment (coocood) [inline executor/executor_xapi.go]: add lock

**Comment by shenli** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
@zimulala

File: executor/executor_xapi.go [hunk 3]
```diff
@@ -196,12 +206,14 @@ func (e *XSelectIndexExec) Next() (*Row, error) {
...
 			limitCount = *e.indexPlan.LimitCount
 		}
+		concurrency := 2
 		if e.indexPlan.NoLimit {
-			e.runTableTasks(len(e.tasks))
+			concurrency = len(e.tasks)
 		} else if limitCount > int64(BaseLookupTableTaskSize) {
-			concurentCount := limitCount/int64(BaseLookupTableTaskSize) + 1
-			e.runTableTasks(int(concurentCount))
+			concurrency = int(limitCount/int64(BaseLookupTableTaskSize) + 1)
```
Comment (zimulala) [inline executor/executor_xapi.go]: Here 2 is the least concurrency value？

Comment (shenli) [inline executor/executor_xapi.go]: if limitCount > int64(BaseLookupTableTaskSize), "limitCount/int64(BaseLookupTableTaskSize" should not be less than 1. So concurrency should not be less than 2.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**